src/abstractdataframe/iteration.jl

```diff
@@ -138,11 +138,17 @@ function Base.iterate(itr::DataFrameColumns{<:AbstractDataFrame,
     return (_names(itr.df)[j] => itr.df[j], j + 1)
 end
 
+# TODO: remove this after deprecation period of getindex of DataFrameColumns
```

Comment:
Why not put this in deprecated.jl?

Reply:
It will be simpler for me to clean it up later (and I assume I will do it just after the coming release).
deprecated.jl has 62kb and almost 2000 loc now and actually it is easy to forget to fix something there.